libs/wire-api-federation/.hspec

```diff
@@ -0,0 +1 @@
+--fail-on-focused
```

Comment:
Isn't there a better solution that applies to all packages?  I'm trying in https://github.com/wireapp/wire-server/pull/1489, but no definite idea yet.

anyway it's fine to do it here now, better than nothing.  (unless somebody will expect it to be universally enabled and rely on it.)

Reply:
There are four ways:
- Specify it in ~/.hspec: Obviously doesn't work for us
- Specify it in HSPEC_OPTIONS env variable: This variable gets lost when stack runs with nix, but could work for CI
- Specify it in command line args: This doesn't work because not all of our tests are hspec, so passing this in `--test-arguments` for stack would cause other tests to fail.
- Specify it in .hspec at the root of each package: This is what I ended up doing.

Reply:
Hmm, I set it in `stack-deps.nix` and it worked! I have now set it in Makefile too and focused a test, if it works, we can use this. Unfortunately, those who don't use nix or make may commit a focused the test, but the CI should catch it.

Reply:
[It worked](https://bridge-ie-concourse.zinfra.io/teams/main/pipelines/wire-server-pr/jobs/wire-server-compile-alpine/builds/3058#L606eb6ee:2368:2369)!!

```
#9 268.5 wire-api-federation              > spec: unexpected argument `"--fail-on-focused"' from environment variable HSPEC_OPTIONS
#9 268.5 wire-api-federation              > Test suite spec failed
```

Reply:
Hmm, I should read things better!

Reply:
The ubuntu job won't fail because it doesn't use the Makefile :(

Reply:
[Much better](https://bridge-ie-concourse.zinfra.io/teams/main/pipelines/wire-server-pr/jobs/wire-server-compile-alpine/builds/3061#L606ebea7:2548:2563)

```
#7 294.3 wire-api-federation              > Test.Wire.API.Federation.Client
#7 294.3 wire-api-federation              >   Federator.Client
#7 294.3 wire-api-federation              >     should make correct calls to the federator and parse success response correctly FAILED [1]
#7 294.3 wire-api-federation              > 
#7 294.3 wire-api-federation              > Failures:
#7 294.3 wire-api-federation              > 
#7 294.3 wire-api-federation              >   test/Test/Wire/API/Federation/ClientSpec.hs:54:7: 
#7 294.3 wire-api-federation              >   1) Test.Wire.API.Federation.Client.Federator.Client should make correct calls to the federator and parse success response correctly
#7 294.3 wire-api-federation              >        item is focused; failing due to --fail-on-focused
#7 294.3 wire-api-federation              > 
#7 294.3 wire-api-federation              >   To rerun use: --match "/Test.Wire.API.Federation.Client/Federator.Client/should make correct calls to the federator and parse success response correctly/"
#7 294.3 wire-api-federation              > 
#7 294.3 wire-api-federation              > Randomized with seed 819810423
#7 294.3 wire-api-federation              > 
#7 294.3 wire-api-federation              > Finished in 0.1371 seconds
#7 294.3 wire-api-federation              > 1 example, 1 failure
```